.gitmodules

```diff
@@ -149,3 +149,6 @@
 [submodule "third_party/mimalloc"]
 	path = third_party/mimalloc
 	url = https://github.com/microsoft/mimalloc.git
+[submodule "third_party/triton-mathaot"]
+	path = third_party/triton-mathaot
+	url = https://github.com/ROCmSoftwarePlatform/triton.git
```

Comment:
How big is the repo? Can it be perhaps fetch only if building with rocm?

Reply:
@malfet To we have a notion of platform specific sub modules?

Reply:
No, but it feels like one should add an optional step to https://github.com/pytorch/pytorch/blob/85d4708512adfdfe5d819cb043eb7172b819691a/setup.py#L370

Reply:
@malfet More specifically, this is the call order:
```
    check_submodules()
    check_pydep("yaml", "pyyaml")

    build_caffe2(
        version=version,
        cmake_python_library=cmake_python_library,
        build_python=True,
        rerun_cmake=RERUN_CMAKE,
        cmake_only=CMAKE_ONLY,
        cmake=cmake,
    )
```

The `build_caffe2` call will configure CMake via the `cmake.generate` call, which will provide the `cmake.get_cmake_cache_variables()` method for us to query if `USE_ROCM` is `True`. That is how subsequent code in `configure_extension_build` refers to CMake variables via `cmake_cache_vars`.

Reply:
Can't we rely on `USE_ROCM` environment variable that is passed to `setup.py` and later passed to the CMAKE process?

Reply:
> Can't we rely on `USE_ROCM` environment variable that is passed to `setup.py` and later passed to the CMAKE process?

Unfortunately we can't. `USE_ROCM` has been removed from the build instructions a few versions earlier.
We do use it internally, but we cannot ask users to do this without detailed documentation.

P.S., the current sizes of all git submodules is well over 3GiB, if users follow the instructions without any hesitation.

Reply:
> Unfortunately we can't. USE_ROCM has been removed from the build instructions a few versions earlier.

Why not use `os.path.exists("/opt/rocm/include/rocm_version.h")` as a condition for checking out the submodule then?

> P.S., the current sizes of all git submodules is well over 3GiB, if users follow the instructions without any hesitation.

This is not the case, users complain about submodule size and this is what I want to reduce to an extend. And correct me if I'm wrong, but `triton-mathaot` is a temporary workaround, isn't it?

Reply:
> And correct me if I'm wrong, but `triton-mathaot` is a temporary workaround, isn't it?

Yes and No, technically this approach will be the standard flash attention implementation for the foreseeable future.
In terms of deployment, we are going internal review and will release this flash attention library as a separate repo when its done, but this standalone library will keep depending on Triton and thus will not reduce recursive submodule size

Reply:
Mark as resolved since the submodule is removed